src/models.py

```diff
@@ -51,22 +52,44 @@ def create_session(self) -> UserSession:
         session.commit()
         return new_session
 
+    @staticmethod
+    def get_users() -> list[Any]:
+        """Use this method to get all users from users table."""
+        session = session_var.get()
+        return session.query(User).all()
+
+    @staticmethod
+    def get_user_by_credentials(username: str, password: str) -> User | None:
+        """Use this method to fetch a user from users table with a specific username and password."""
+        session = session_var.get()
+        user_to_fetch: User | None = session.query(User).filter_by(username=username).first()
+        if user_to_fetch and user_to_fetch.check_password(password):
+            return user_to_fetch
+        return None
 
-class UserSession(Base):  # pylint: disable=too-few-public-methods
+
+class UserSession(Base):
     """A model class for user_session table."""
 
     __tablename__ = "user_session"
 
     id: int = Column(Integer, primary_key=True)  # noqa: A003
     session_id: str = Column(String, nullable=False, unique=True)
     user_id: int = Column(Integer, ForeignKey("users.id"), nullable=False)
+    user: User = relationship("User", uselist=False)
 
     def delete(self) -> None:
         """Use this method to delete a user session."""
         session = session_var.get()
         session.delete(self)
         session.commit()
 
+    @staticmethod
+    def get_user_session_by_session_id(session_id: str | None) -> UserSession | None:
```

Comment:
Seems like `session_id` parameter should be strictly `int`. Otherwise it seems a little bit confusing - there is no sessions with `session_id=None`, so `get_user_session_by_session_id(None)` has no sense.

I probably know why you made this parameter to be optional, it's because of this code:
```python
session_id = request.cookies.get("session_id")
user_session = UserSession.get_user_session_by_session_id(session_id)
```
Here `request.cookies.get()` can return `None` and mypy complained about `session_id` parameter in `get_user_session_by_session_id`.

To solve this problem we can make a little refactoring of the previous code to this:
```python
session_id = request.cookies.get("session_id")

user_session = None
if session_id is not None:
    user_session = UserSession.get_user_session_by_session_id(session_id)
```

It will solve the mypy issue and allow us to make `session_id` parameter to be sctrictly `int`.